ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -112,7 +112,10 @@ public ExpressionMetadata buildCodeGenFromParseTree(
       final SqlType expressionType = expressionTypeManager
           .getExpressionSqlType(expression);
 
-      ee.setExpressionType(SQL_TO_JAVA_TYPE_CONVERTER.toJavaType(expressionType));
+      if (expressionType != null) {
+        // expressionType can be null if expression is NULL.
+        ee.setExpressionType(SQL_TO_JAVA_TYPE_CONVERTER.toJavaType(expressionType));
+      }
```

Comment:
This avoids an NPE.  If not called the return value is left at the default `Object`, which is fine for nulls.